host_core/lib/host_core/control_interface/server.ex

```diff
@@ -0,0 +1,218 @@
+      start_provider_command = Jason.decode!(body)
+
+      ack =
+        case HostCore.Providers.ProviderSupervisor.start_executable_provider_from_oci(
+               start_provider_command["provider_ref"],
+               start_provider_command["link_name"]
+             ) do
+          {:ok, _pid} ->
+            %{
+              host_id: host_id,
+              provider_ref: start_provider_command["provider_ref"]
+            }
+
+          {:error, e} ->
+            %{
+              host_id: host_id,
+              provider_ref: start_provider_command["provider_ref"],
+              error: "Failed to start provider: #{e}"
+            }
+        end
+
+      IO.inspect(ack)
+      Gnat.pub(:control_nats, reply_to, Jason.encode!(ack))
+    end
+  end
+
+  # Stop Provider
+  defp handle_request({"cmd", host_id, "sp"}, body, reply_to) do
+    if host_id == HostCore.Host.host_key() do
+      stop_provider_command = Jason.decode!(body)
+
+      HostCore.Providers.ProviderSupervisor.terminate_provider(
+        stop_provider_command["provider_ref"],
+        stop_provider_command["link_name"]
+      )
+
+      Gnat.pub(:control_nats, reply_to, Jason.encode!(%{}))
+    end
+  end
+
+  # Update Actor
+  defp handle_request({"cmd", host_id, "upd"}, body, reply_to) do
+    if host_id == HostCore.Host.host_key() do
+      update_actor_command = Jason.decode!(body)
+
+      # TODO - live updates for actors is not implemented yet
+    end
+  end
+
+  # Auction Actor
+  # input: #{"actor_ref" => "...", "constraints" => %{}}
+  defp handle_request({"auction", "actor"}, body, reply_to) do
+    auction_request = Jason.decode!(body)
+    host_labels = HostCore.Host.host_labels()
+    required_labels = auction_request["constraints"]
+
+    if Map.equal?(host_labels, Map.merge(host_labels, required_labels)) do
+      ack = %{
+        actor_ref: auction_request["actor_ref"],
+        constraints: auction_request["constraints"],
+        host_id: HostCore.Host.host_key()
+      }
+
+      Gnat.pub(:control_nats, reply_to, Jason.encode!(ack))
+    end
+  end
+
+  # Auction Provider
+  # input: #{"actor_ref" => "...", "constraints" => %{}}
+  defp handle_request({"auction", "provider"}, body, reply_to) do
+    auction_request = Jason.decode!(body)
+    host_labels = HostCore.Host.host_labels()
+    required_labels = auction_request["constraints"]
+
+    # TODO - don't answer this request if we're already running a provider
+    # that matches this link_name and ref.
+    if Map.equal?(host_labels, Map.merge(host_labels, required_labels)) do
+      ack = %{
+        provider_ref: auction_request["provider_ref"],
+        link_name: auction_request["link_name"],
+        constraints: auction_request["constraints"],
+        host_id: HostCore.Host.host_key()
+      }
+
+      Gnat.pub(:control_nats, reply_to, Jason.encode!(ack))
+    end
+  end
+
+  # FALL THROUGH
+  defp handle_request(tuple, _body, _reply_to) do
+    IO.puts("Got here #{tuple}")
+  end
```

Comment:
I'm fine with an "unsupported request" but I think it would be better as `Logger.warn`